torch/csrc/jit/passes/quantization.cpp

```diff
@@ -164,12 +179,15 @@ Node* InsertObserversHelper::insertObserverFor(
   } else {
     observer_module = std::get<0>(qconfig);
   }
-  std::string observer_name = "observer_for_" + v->debugName();
   script::Module observer = observer_module.clone();
+  std::string observer_name = "_observer_" + std::to_string(uid_.getUniqueId());
```

Comment:
Why can't we just have `size_t uid = 0;` here and `uid++;` in the loop? What's the benefit of having a common enumeration across different uses? The check `module.find_module(observer_name)` is sufficient for correctness - if it passes then there are no issues even if this `uid` overlaps with `uid` used in another module.

Reply:
you mean always start from 0? then we are doing N^2 checks?

Reply:
Frankly I think even N^2 won't be a problem here because number of observers won't be that big while readability of the code would improve by not introducing a new class. Also, you could move `size_t uid = 0;` to the beginning of this function and thus make sure that we're not rechecking the same `uid` more than once per method of the module - in real life I think it would mean that we're almost never rechecking them, so you get the O(N) complexity.

Reply:
I'm not sure I follow, this function is adding observer for one Value, how is moving `uid` to the beginning of this function prevent N^2 checks? Are you saying moving this outside of this function?

Reply:
I think we can remove the class by changing UniqueId to `int`